### PR TITLE
provider/aws: CodeCommit - Currently only in us-east-1

### PR DESCRIPTION
**Pull request by stack72**
- [x] Schema
- [x] Resource
- [x] Acceptance Tests
- [x] Docs

In order to run the acceptance tests, we need to set an ENV_VAR for Region to be us-east-1

then we get the following results:

```
ake testacc TEST=./builtin/providers/aws TESTARGS='-run=CodeCommitRepository' 2>~/tf.log
go generate ./...
TF_ACC=1 go test ./builtin/providers/aws -v -run=CodeCommitRepository -timeout 90m
=== RUN TestAccAWSCodeCommitRepository_basic
--- PASS: TestAccAWSCodeCommitRepository_basic (5.39s)
=== RUN TestAccAWSCodeCommitRepository_withChanges
--- PASS: TestAccAWSCodeCommitRepository_withChanges (5.59s)
PASS
ok  	github.com/hashicorp/terraform/builtin/providers/aws	10.998s
```

### CODE REVIEW

**Comment by stack72** [comment]
@radeksimko @phinze I have just rebased this PR. *Technically*, this is now complete as CodeCommit is only available in us-east-1 right now

Would love to know your thoughts

File: builtin/providers/aws/config.go [hunk 1]
```diff
@@ -213,6 +215,9 @@ func (c *Config) Client() (interface{}, error) {
 
 		log.Println("[INFO] Initializing CodeDeploy Connection")
 		client.codedeployconn = codedeploy.New(sess)
+
+		log.Println("[INFO] Initializing CodeCommit SDK connection")
+		client.codecommitconn = codecommit.New(awsConfig)
```
Comment (phinze) [inline builtin/providers/aws/config.go]: Hm - did codecommit not get the new session.Session refactoring that the rest of the connections did this week? I'm surprised this builds!

I'd expect the line here to need `usEast1Sess` as the argument.

Comment (stack72) [inline builtin/providers/aws/config.go]: you are 100% correct. I have just updated this commit. It should indeed be the sess work

I'm also surprised this built

**Comment by stack72** [comment]
This has been updated to use the new sess based refactoring

File: builtin/providers/aws/resource_aws_codecommit_repository.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+			"default_branch": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsCodeCommitRepositoryCreate(d *schema.ResourceData, meta interface{}) error {
+	codecommitconn := meta.(*AWSClient).codecommitconn
```
Comment (phinze) [inline builtin/providers/aws/resource_aws_codecommit_repository.go]: Stylistically we've tended to call the client just `conn` when it's a local variable.

Comment (phinze) [inline builtin/providers/aws/resource_aws_codecommit_repository.go]: (Unless of course you're juggling two in a given function.)

**Comment by phinze** [comment]
This looks great! Merging as is and I'll fix that style nit on master.

**Comment by stack72** [comment]
Wow, you are on fire tonight! thanks :)

**Pull request merged**